### PR TITLE
Validate Client callback_url against Client redirect_uris

### PR DESCRIPTION
**Pull request by OptimusRhine**
https://www.pivotaltracker.com/story/show/149047283

* added new `Client.callback_url` validation on `/client/<client_id> [POST]`
  * `callback_url`'s `{scheme}://{netloc}` must match that of any of the provided `redirect_urls`
  * only triggers if `callback_url` is provided (i.e. the field is still optional)
* modified `test_auth.test_client_edit`, so that client's new `callback_url` is also present in its list of `redirect_urls`

### CODE REVIEW

File: portal/views/auth.py [hunk 1]
```diff
@@ -571,6 +572,16 @@ def validate_application_origins(form, field):
             if not url_validation(url, require_tld=False):
                 raise validators.ValidationError("Invalid URL")
 
+    def validate_callback_url(form, field):
```
Comment (ivan-c) [inline portal/views/auth.py]: Could you add a docstring here, describing what this validation function does?

File: tests/test_auth.py [hunk 2]
```diff
@@ -65,15 +65,17 @@ def test_client_bad_add(self):
     def test_client_edit(self):
```
Comment (ivan-c) [inline tests/test_auth.py]: Could you add another test, testing that submitting a CB url with a different domain does in fact cause an error (regression test)?

**Comment by OptimusRhine** [comment]
@ivan-c Thanks Ivan! Added description to that new validation function, and added a new negative test.

File: tests/test_auth.py [hunk 3]
```diff
@@ -65,15 +65,28 @@ def test_client_bad_add(self):
...
-        self.assertEquals(client.callback_url, 'http://tryme.com')
+        self.assertEquals(client.callback_url, test_url)
+
+        invalid_url = "http://invalid.org"
+        rv2 = self.client.post('/client/{0}'.format(client.client_id),
+                data=dict(callback_url=invalid_url,
+                          application_origins=origins,
+                          application_role=INTERVENTION.DEFAULT.name))
+        # 200 response, because page is reloaded with validation errors
+        self.assert200(rv2)
```
Comment (ivan-c) [inline tests/test_auth.py]: Could you make the assertion more specific, eg check for the specific error in the reponse (`rv2.data`)?

**Pull request merged**